subprojects/core/src/main/java/org/gradle/configuration/ScriptPluginFactorySelector.java

```diff
@@ -69,6 +84,15 @@ private ScriptPluginFactory scriptPluginFactoryFor(String fileName) {
     }
 
     private ScriptPluginFactory findScriptPluginFactoryFor(String fileName) {
+        for (ScriptingLanguage scriptingLanguage: scriptingLanguages) {
+            if (fileName.endsWith("." + scriptingLanguage.getExtension())) {
```

Comment:
What if we changed `scriptingLanguage.getExtension()` contract to include the leading dot as to avoid this string concatenation and declutter the code a bit?